src/index.js

```diff
@@ -114,15 +117,27 @@ function exit(signal) {
   }, 500);
 }
 
+async function startService() {
+  if(!await qclient.isConnected()){
+    logger.info('qtum is still starting, please wait');
+    setTimeout(()=>{
+      startService();
+    },1000);
+  }else{
+    startSync();
+    startAPI();
+    openBrowser();
+  }
+}
+
 process.on('SIGINT', exit);
 process.on('SIGTERM', exit);
 process.on('SIGHUP', exit);
 
 startQtumProcess(false);
 
-// Wait 4s for qtumd to start and reindex if necessary
-setTimeout(() => {
-  startSync();
-  startAPI();
-  openBrowser();
-}, 5000);
+//// Wait 4s for qtumd to start and reindex if necessary
+setTimeout(()=>{
+  startService();
```

Comment:
no need to use timeout here. lets just automatically start checking if qtumd is running. that's the whole point of the repeating logic.